sources/packages/backend/apps/queue-consumers/src/processors/schedulers/workflow/assessment-workflow-enqueuer.scheduler.ts

```diff
@@ -41,14 +41,31 @@ export class AssessmentWorkflowEnqueuerScheduler extends BaseScheduler<void> {
       processSummary.info(
         "Checking application assessments to be queued for start.",
       );
-      // Process summary to be populated by the enqueueStartAssessmentWorkflows.
+      // Process summary to be populated by each enqueueing workflow call.
       // In case an unexpected error happen the finally block will still be able to
       // output the partial information captured by the processSummary.
-      const serviceProcessSummary = new ProcessSummary();
-      processSummary.children(serviceProcessSummary);
-      await this.workflowEnqueuerService.enqueueStartAssessmentWorkflows(
-        serviceProcessSummary,
-      );
+      try {
+        const serviceProcessSummary = new ProcessSummary();
+        processSummary.children(serviceProcessSummary);
+        await this.workflowEnqueuerService.enqueueCancelAssessmentWorkflows(
+          serviceProcessSummary,
+        );
+      } catch (error: unknown) {
+        const errorMessage =
+          "Unexpected error while enqueueing cancel assessment workflows.";
+        processSummary.error(errorMessage, error);
+      }
+      try {
```

Comment:
An alternative way to reduce the try/catch nesting would be maybe a refactor like below for the `enqueueAssessmentOperations` method.

```ts
async enqueueAssessmentOperations(job: Job<void>): Promise<string[]> {
  const processSummary = new ProcessSummary();
  try {
    processSummary.info(
      "Checking application assessments to be queued for start.",
    );
    // Check for applications with assessments to be cancelled.
    await this.executeEnqueueProcess(
      processSummary,
      this.workflowEnqueuerService.enqueueCancelAssessmentWorkflows,
    );
    // Check for applications with assessments to be started.
    await this.executeEnqueueProcess(
      processSummary,
      this.workflowEnqueuerService.enqueueStartAssessmentWorkflows,
    );
    return getSuccessMessageWithAttentionCheck(
      "Process finalized with success.",
      processSummary,
    );
  } catch (error: unknown) {
    const errorMessage = "Unexpected error while executing the job.";
    processSummary.error(errorMessage, error);
    return [errorMessage];
  } finally {
    this.logger.logProcessSummary(processSummary);
    await logProcessSummaryToJobLogger(processSummary, job);
    await this.cleanSchedulerQueueHistory();
  }
}

private async executeEnqueueProcess(
  parentProcessSummary: ProcessSummary,
  enqueueProcess: (summary: ProcessSummary) => Promise<void>,
): Promise<void> {
  try {
    // Process summary to be populated by each enqueueing workflow call.
    // In case an unexpected error happen the finally block will still be able to
    // output the partial information captured by the processSummary.
    const serviceProcessSummary = new ProcessSummary();
    parentProcessSummary.children(serviceProcessSummary);
    await enqueueProcess(serviceProcessSummary);
  } catch (error: unknown) {
    const errorMessage =
      "Unexpected error while enqueueing start assessment workflows.";
    parentProcessSummary.error(errorMessage, error);
  }
}
```

Reply:
Apparently it looks good and TS doesn't complain but it doesn't work. I'm going to investigate why further when possible.
It seems to be this kind of problem: https://stackoverflow.com/questions/62989503/this-is-undefined-when-function-of-an-inject-able-class-is-sent-as-argument-to

Reply:
It is probably the case to have the parameter changed to be received as 
`(...) =>this.workflowEnqueuerService.enqueueStartAssessmentWorkflows(...)`.